### PR TITLE
panw: process URLs for THREAT events of url subtype

### LINKED ISSUE #4711: [PANW] url.original is not parsed
The PANW integration may have an `url.original` but this field is not parsed in its subcomponents, e.g., `url.domain` as in other integrations (e.g., `Zscaler_zia.web`)

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This processes the PANW URL/Filename (misc) field to either a url object with a uri_parts processor if the type is a THREAT and subtype url. It puts the misc field into file.path or file.name base on a slash heuristic.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Closes #4711

### CODE REVIEW

File: packages/panw/data_stream/panos/elasticsearch/ingest_pipeline/threat.yml [hunk 1]
```diff
@@ -287,8 +287,22 @@ processors:
...
+      if: ctx.panw?.panos?.sub_type == 'file' && (ctx.panw.panos.misc.contains('/') || ctx.panw.panos.misc.contains('\\'))
+      field: file.path
+      copy_from: panw.panos.misc
+      ignore_failure: true
   - set:
-      field: url.original
+      if: ctx.panw?.panos?.sub_type == 'file' && !(ctx.panw.panos.misc.contains('/') || ctx.panw.panos.misc.contains('\\'))
+      field: file.name
       copy_from: panw.panos.misc
       ignore_failure: true
```
Comment (kcreddy) [inline packages/panw/data_stream/panos/elasticsearch/ingest_pipeline/threat.yml:307]: Here, both `file.name` and `file.path` seems to assume same value `panw.panos.misc`. Should we instead split by `/` or `\\` and capture the last part into `file.name`?

**Review by kcreddy — APPROVED** [review]
Just minor clarification. LGTM 👍🏼

File: packages/panw/data_stream/panos/elasticsearch/ingest_pipeline/threat.yml (hunk 1 above)
Comment (efd6) [reply packages/panw/data_stream/panos/elasticsearch/ingest_pipeline/threat.yml:307]: Yeah, that's probably a good idea. I'll do that.

**Review by kcreddy — APPROVED** [review]
LGTM

File: packages/panw/data_stream/panos/elasticsearch/ingest_pipeline/threat.yml (hunk 1 above)
Comment (kcreddy) [reply packages/panw/data_stream/panos/elasticsearch/ingest_pipeline/threat.yml]: LGTM

**Pull request merged**